src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -977,6 +978,322 @@ def resolve_imports_and_packages(
     )
 
 
+def resolve_packages(
+    session: "snowflake.snowpark.Session",
+    packages: Optional[List[Union[str, ModuleType]]],
+    is_pandas_udf: bool = False,
+    statement_params: Optional[Dict[str, str]] = None,
+) -> str:
+
+    # resolve packages
+    resolved_packages = (
+        session._resolve_packages(
+            packages, include_pandas=is_pandas_udf, statement_params=statement_params
+        )
+        if packages is not None
+        else session._resolve_packages(
```

Comment:
`_resolve_packages` includes calls to Snowflake for getting `information_schema.packages` view and for fetching and/or uploading packages to a Snowflake stage. With the CLI local sandbox, we want to short circuit this and exit before interaction with Snowflake. We only want to parse packages and check them against the local environment of the user, and return from there. This is because we are not registering anything with Snowflake and hence do not need to resolve any of the packages, supported or unsupported.